From 8cfae1a0c5e599c62cdae6474b257b3763cb0d8a Mon Sep 17 00:00:00 2001
From: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
Date: Wed, 26 Apr 2017 16:24:25 +0200
Subject: [PATCH 08/10] arm64: add ESR_ELx_SYS64_ISS_SYS_OP_MASK define

Patch-mainline: Never, partially adding already mainlined patch
References: bsc#1035997

6126ce0588eb arm64: Add CNTVCT_EL0 trap handler

needs ESR_ELx_SYS64_ISS_SYS_OP_MASK define which is added by
following patch:

116c81f427ff arm64: Work around systems with mismatched cache line sizes

However, there is a long dependency chain to bring in this patch. Adding
just the required define instead.

Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 arch/arm64/include/asm/esr.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/include/asm/esr.h b/arch/arm64/include/asm/esr.h
index f8f837319eb0..5c16909a704a 100644
--- a/arch/arm64/include/asm/esr.h
+++ b/arch/arm64/include/asm/esr.h
@@ -149,6 +149,9 @@
 					 ((op2) << ESR_ELx_SYS64_ISS_OP2_SHIFT) | \
 					 ((crn) << ESR_ELx_SYS64_ISS_CRN_SHIFT) | \
 					 ((crm) << ESR_ELx_SYS64_ISS_CRM_SHIFT))
+
+#define ESR_ELx_SYS64_ISS_SYS_OP_MASK	(ESR_ELx_SYS64_ISS_SYS_MASK | \
+					 ESR_ELx_SYS64_ISS_DIR_MASK)
 /*
  * User space cache operations have the following sysreg encoding
  * in System instructions.
-- 
2.11.0

